src/kerberos_client.py

```diff
@@ -23,58 +100,120 @@ def register(self, username: str):
+                    "code": 1024,
+                    "payloadSize": len(username) + len(password)
+                },
+                "payload":{
+                    "name": username,
+                    "password": password
+                }
+            }
+            uuid = authserver.register(request)["Payload"]
+            self.sha256 = create_password_sha(password)
+        finally:
+            with open(CLIENT_FILE, 'w') as client_file:
+                client_file.writelines([username, uuid])
 
 
     def receive_aes_key(self):
         """
         receives an aes key and a ticket to a message server.
         decrypts the key and saves it along with the ticket for future use.
         """
-        nonce = create_nonce()
-        encrypted_key, ticket = authserver.generate_session_key(self.uuid, SERVER_ID, nonce)
         try:
-            decrypted_key = decrypt_aes(encrypted_key, self.sha256)
-            self.aes_key = decrypted_key
-            self.ticket = ticket
-        except ValueError as e:
-            print(e)
-            print(ERROR_MESSAGE)
+            client_info = self.get_client_info()
+            uuid = client_info["uuid"]
+            nonce = create_nonce()
+            request = {
+                "header":{
+                    "clientID": uuid,
+                    "version": self.version,
+                    "code": 1027,
+                    "payloadSize": len(SERVER_ID) + len(nonce)
+                },
+                "payload":{
+                    "serverID": SERVER_ID,
+                    "nonce": nonce
+                }
+            }
+            encrypted_key, ticket = authserver.generate_session_key(request)
+            try:
+                decrypted_key = str(decrypt_aes(encrypted_key, self.sha256))
+                self.aes_key = decrypted_key
+                self.ticket = ticket
+            except ValueError as e:
+                print(e)
+                print(ERROR_MESSAGE)
+        except FileNotFoundError:
+            print("File 'me.info' was not found.")
+            exit(1)
             
 
     def send_aes_key(self, aes_key):
         """
         sends an authenticator and a ticket to the message server.
         :param aes_key: AES Symmetric Key.
         """
-        pass
+        try:
+            uuid = self.get_client_info()["uuid"]
+            authenticator, authenticatorSize = self.create_authenticator(uuid)
+            ticket = self.ticket
+            request = {
+                "header": {
+                    "clientID": uuid,
+                    "version": self.version,
+                    "code": 1028,
+                    "payloadSize": authenticatorSize + len(ticket)
+                },
+                "payload":{
+                    "authenticator": authenticator,
+                    "ticket": ticket
+                }
+            }
+            msgserver.receive_aes_key(request)
+        except FileNotFoundError:
```

Comment:
`FileNotFoundError` is one error but we can experience more, for now change it to more general error (or add a more general error beneath) so code won't fail